Snakefile

```diff
@@ -304,11 +305,51 @@ rule differential_gsea:
         else
             touch {wildcards.accession}.{wildcards.contrast_id}.{wildcards.ext_db}.gsea.tsv
             touch {wildcards.accession}.{wildcards.contrast_id}.{wildcards.ext_db}.gsea_list.tsv
-            touch {wildcards.accession}.{wildcards.contrast_id}.{wildcards.ext_db}.gsea.tsv_WARNING_not_computed
-            touch {wildcards.accession}.{wildcards.contrast_id}.{wildcards.ext_db}.gsea_list.tsv_WARNING_not_computed
         fi
         """
 
+
+rule check_differential_gsea:
+    """
+    Whether there is a lack of annotation for 'ext_db' in rule differential_gsea, or the
+    analysis produces no results passing significance cutoff, gsea output files could be empty.
+    Here we aim to delete those empty files.
+    """
+    log: "logs/{accession}.{contrast_id}.{ext_db}-check-differential_gsea.log"
+    input:
+	    gsea="{accession}.{contrast_id}.{ext_db}.gsea.tsv",
+        gsea_list="{accession}.{contrast_id}.{ext_db}.gsea_list.tsv"
+    output:
+	    log1="logs/{accession}.{contrast_id}.{ext_db}.gsea.tsv",
+        log2="logs/{accession}.{contrast_id}.{ext_db}.gsea_list.tsv"
+    shell:
```

Comment:
There are some misalignments on the indents here, please fix. Also, I would just leave all these extra logging lines on the main log file of this rule, or did you have any reason not to do it like that? If you need an output file to be generated, there is some dummy or temp mechanism and you can just make a `touch check_differential_gsea.done` in the logs folder, which will get deleted if it is a temp.

Reply:
Thanks, very useful. Please see commits below - primarily 3ba9d1f